src/Interpreters/executeQuery.cpp

```diff
@@ -419,6 +420,13 @@ void logQueryFinish(
         {
             Progress p;
             p.incrementPiecewiseAtomically(Progress{ResultProgress{elem.result_rows, elem.result_bytes}});
+
+
+            if (info.profile_counters)
+            {
+                UInt64 real_time_microseconds = (*info.profile_counters)[ProfileEvents::RealTimeMicroseconds];
+                p.incrementRealTimeMicroseconds(real_time_microseconds);
+            }
```

Comment:
This is not very good solution to have this dependent from irrelevant `log_profile_events` setting. Let's replace line 402 with:
```
        std::shared_ptr<ProfileEvents::Counters::Snapshot> profile_counters;

        QueryStatusInfo info = process_list_elem->getInfo(true, true);
        if (context->getSettingsRef().log_profile_events)
            profile_counters = info.profile_counters;
        else
            profile_counters.swap(info.profile_counters);
```
and this code with:
```
            p.incrementRealTimeMicroseconds(profile_counters[ProfileEvents::RealTimeMicroseconds]);
```

Reply:
Thanks for the suggestion